src/components/Personalization/createExecuteDecisions.js

```diff
@@ -20,6 +20,31 @@ const buildActions = decision => {
   return decision.items.map(item => assign({}, item.data, { meta }));
 };
 
+const dedupe = metas => {
+  const result = [];
+  const set = new Set();
+
+  metas.forEach(item => {
+    if (set.has(item.id)) {
+      return;
+    }
+
+    set.add(item.id);
+    result.push(item);
+  });
+
+  return result;
+};
+
+const collectMetas = (collect, decisionMetas) => {
+  const metas = flatMap(decisionMetas, identity);
+  const dedupedMetas = dedupe(metas.map(e => e.meta));
```

Comment:
I'm a little confused by the `metas` naming, because I would expect each item in `metas` to be a `meta`, but it seems like it's an object with a `meta` property. This might be something to improve is something comes to mind for you.